File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/common/MutedException.java
package solutions.digamma.damas.jcr.common;

/**
 * Muted exception.
 *
 * @author Ahmad Shahwan
 */
public class MutedException extends RuntimeException {

    /**
     * Cause as an {@link Exception}.
     */
    private Exception cause;

    MutedException(String message, Exception cause) {
        super(message, cause);
        this.cause = cause;
    }

    MutedException(Exception cause) {
        this("Exception muted.", cause);
    }

    @Override
    public Exception getCause() {
        return this.cause;
    }
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/common/MutedException.kt
package solutions.digamma.damas.jcr.common

/**
 * Muted exception.
 *
 * @author Ahmad Shahwan
 */
class MutedException internal constructor(
        message: String,
        /**
        * Cause as an [Exception].
        */
        override val cause: Exception) : RuntimeException(message, cause) {

    internal constructor(cause: Exception) : this("Exception muted.", cause)
}
